op-batcher/batcher/espresso.go

```diff
@@ -45,42 +44,43 @@ func (l *BatchSubmitter) tryPublishBatchToEspresso(ctx context.Context, transact
 	ticker := time.NewTicker(transactionFetchInterval)
 	defer ticker.Stop()
 
-Loop:
 	for {
 		select {
 		case <-ticker.C:
-			_, err = l.Espresso.FetchTransactionByHash(ctx, txHash)
+			txinfo, err := l.Espresso.FetchTransactionByHash(ctx, txHash)
 			if err == nil {
-				break Loop
+				l.Log.Warn("Tx included in block ", "txhash", txHash, "blockNr", txinfo.BlockHeight)
+				return nil
 			}
 		case <-timer.C:
 			l.Log.Error("Failed to fetch transaction by hash after multiple attempts", "txHash", txHash)
 			return fmt.Errorf("failed to fetch transaction by hash: %w", err)
 		case <-ctx.Done():
 			l.Log.Info("Cancelling transaction publishing", "txHash", txHash)
-			break Loop
+			return nil
 		}
 	}
-
-	return nil
 }
 
 // Converts a block to an EspressoBatch and starts a goroutine that publishes it to Espresso
 // Returns error only if batch conversion fails, otherwise it is infallible, as the goroutine
 // will retry publishing until successful.
 func (l *BatchSubmitter) queueBlockToEspresso(ctx context.Context, block *types.Block) error {
-	batch, _, err := derive.BlockToSingularBatch(l.RollupConfig, block)
+	l.blockMutex.Lock()
+	l.blocks[block.Hash()] = block
```

Comment:
Can be removed

Reply:
Removed in https://github.com/EspressoSystems/optimism-espresso-integration/pull/96/commits/99e32de72a7e322476c2183523d45f47cf0d014c.